tgg/research/gnomad_tgps.py

```diff
@@ -52,111 +62,125 @@
     mt = mt.annotate_cols(s_in_pangenome=pangenome_ids.contains(mt.s))
-    pg_mt = mt.filter_cols(mt.s_in_pangenome)
     mt = mt.filter_cols(~mt.s_in_pangenome)
 
+    # Annotate mt with allele frequencies and filter out rare SNVs (AF lower than 1%).
+    # Also filter out sites where no samples had a variant call.
+    mt = mt.annotate_rows(freq=gnomad_freq_ht[mt.row_key].freq)
+    mt = mt.filter_rows(mt.freq.AF[0] > 0.01)
+    mt = mt.filter_rows(hl.agg.any(mt.GT.is_non_ref()))
+
+    # Read the HGDP+1KG subset matrix table and split multi-allelics.
+    pg_mt = hl.read_matrix_table(HGDP_SUBSET_MT)
+    pg_mt = hl.experimental.sparse_split_multi(pg_mt)
+
+    # Filter pg_mt to only contain gnomAD samples in pangenome.
+    # Also filter out sites where no samples had a variant call.
+    pg_mt = pg_mt.filter_cols(hl.is_defined(meta_ht[pg_mt.col_key]))
+    pg_mt = pg_mt.filter_cols(pangenome_ids.contains(pg_mt.s))
+    pg_mt = pg_mt.filter_rows(hl.agg.any(pg_mt.GT.is_non_ref()))
+
     # Log the numbers of gnomAD samples not in and in the pangenome reference.
     num_in_samples = pg_mt.count_cols()
     num_out_of_samples = mt.count_cols()
-    logger.info("The number of samples both in pangenome and gnomAD v3 is: ",
+    logger.info("The number of samples both in pangenome and gnomAD v3 is: %i",
                 num_in_samples)
-    logger.info("The number of samples only in gnomAD v3 is: ", num_out_of_samples)
+    logger.info("The number of samples only in gnomAD v3 is: %i", num_out_of_samples)
 
-    # Annotate the gnomAD matrix table with allele frequencies and
-    # filter out SNVs with minor allele frequencies lower than 1% and
-    # filter out samples with no variant call.
-    mt = mt.annotate_rows(freq=gnomad_freq_ht[mt.row_key].freq)
-    mt = mt.filter_rows(mt.freq.AF[1] > 0.01)  # ##why AF[1]?
-    mt = mt.filter_rows(hl.agg.any(mt.GT.is_non_ref()))
+    # Checkpoint the rows from the pangenome MT if no checkpoint file exists.
+    pg_ht = pg_mt.rows()
+    if not hfs.is_file(PANGENOME_CHECKPOINT_FILEPATH):
+        pg_ht.checkpoint(PANGENOME_CHECKPOINT_FILEPATH)
 
-    # Filter out samples with no variant call in the pangenome matrix table.
-    pg_mt = pg_mt.filter_rows(hl.agg.any(pg_mt.GT.is_non_ref()))
-
-    # Checkpoint the rows from the pangenome MT.
-    # Issue: the job is still quite large with 115376 partitions.
-    # logger.info("The number of variants for the pangenome data set is ",
-    # pg_mt.count_rows())
+    logger.info("The number of variants for the pangenome data set is %i",
+                pg_ht.count())
 
     # Annotate the gnomAD matrix table to count the number of minor alleles carried
     # by each participant.
     mt = mt.annotate_cols(total_var_per_sample=hl.agg.count_where(mt.GT.is_non_ref()))
 
     # Filter out SNVs that are in the reference group.
-    mt_filtered = mt.anti_join_rows(pg_mt.rows())
+    pg_ht = hl.read_table(PANGENOME_CHECKPOINT_FILEPATH,
```

Comment:
you should move the `pg_ht.count()` to be after this read, since the current position of the count means that all of the upstream compute gets run twice (once on the checkpoint and once on the count)